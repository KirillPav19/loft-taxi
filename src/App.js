import React, { useState } from 'react';
import {LoginForm} from './Login/index';
import {SignUpForm} from './Signup/index';
import {MapForm} from './Map/index';
import {ProfileForm} from './Profile/index';
import {Header} from './shared/header';

const AppContext = React.createContext({})
export const AppProvider = AppContext.Provider
export const AppConsumer = AppContext.Consumer

const getPage = (page, setPage) => {
    switch (page) {
        case "Login":
            return (
                <LoginForm setPage={setPage}
                           name={'Login'}
                           navigatePage={'Map'}
                />
            )
        case "SignUp":
            return (
                <SignUpForm setPage={setPage}
                            name={'Sign Up'}
                            navigatePage={'Login'}

                />
            )
        case "Map":
            return (
                <MapForm name={'Map'}/>
            )
        case "Profile":
            return (
                <ProfileForm name={'Profile'}/>
            )
        default:
            return (
                <LoginForm setPage={setPage}
                           name={'Login'}
                           navigatePage={'Map'}
                />
            )
    }
}

function App() {
    const [page, setPage] = useState('Login')
    const [isLoggedIn, setIsLoggedIn] = useState(false)

    const login = (email, password) => {
        if (email === 'loft@.ru' && password === '1234') {
            setIsLoggedIn(true)
            setPage('Map')
        }
    }

    const logout = () => {
        setIsLoggedIn(false)
        setPage('Login')
    }
    return (
        <AppProvider value={{login, logout, isLoggedIn}}>
            <div>
                <Header setPage={setPage} id={'header'}/>
                {getPage(page, setPage)}
            </div>
        </AppProvider>
    );
}

export default App;
