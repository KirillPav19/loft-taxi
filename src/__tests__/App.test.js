import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import '../setupTests'

describe('App', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders Header', () => {
        const wrapper = shallow(<App/>);
        expect(wrapper.find('#header').length === 1).toEqual(true);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

})

// FIX TypeError: window.URL.createObjectURL is not a function
jest.mock('mapbox-gl', () => ({
    Map: () => ({})
}))