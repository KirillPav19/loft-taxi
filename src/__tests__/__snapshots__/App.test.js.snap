// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    className="Style-container-button"
  >
    <button
      className="Style-button"
      onClick={[Function]}
    >
      Login
    </button>
    <button
      className="Style-button"
      onClick={[Function]}
    >
      SingUp
    </button>
    <button
      className="Style-button"
      onClick={[Function]}
    >
      Map
    </button>
    <button
      className="Style-button"
      onClick={[Function]}
    >
      Profile
    </button>
    <button
      className="Style-button"
      onClick={[Function]}
    >
      Exit
    </button>
  </div>
  <form
    onSubmit={[Function]}
  >
    <h1>
      LOGIN
    </h1>
    <div>
      <input
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </div>
    <div>
      <input
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <button
      className="Style-button"
      onClick={[Function]}
    >
      Login
    </button>
  </form>
</div>
`;
